sdk/core/Azure.Core/src/Shared/ProtocolOperation.cs

```diff
@@ -30,6 +30,17 @@ internal ProtocolOperation(ClientDiagnostics clientDiagnostics, HttpPipeline pip
         public override string Id => throw new NotSupportedException();
 #pragma warning restore CA1822
 
+        /// <inheritdoc />
+        public override RehydrationToken? GetRehydrationToken()
+        {
+            if (_nextLinkOperation is NextLinkOperationImplementation nextLinkOperation)
+            {
+               return nextLinkOperation?.GetRehydrationToken();
+            }
+
+            return null;
+        }
```

Comment:
Supposedly `_nextLinkOperation` will always be a `NextLinkOperationImplementation` in data-plane scenarios. I didn't want to force a cast here to avoid throwing in unforeseen scenarios. Before this change, `GetRehydrationToken` would always return `null` anyway.